tests/lib/rules/no-unused-vars.js

```diff
@@ -333,6 +345,21 @@ ruleTester.run("no-unused-vars", rule, {
             ]
         },
 
+        // Destructured var without object rest sibling.
+        {
+            code: "const data = { type: 'coords', x: 1, y: 2 };\nconst { type } = data;",
+            options: [{ ignoreRestSiblings: true }],
+            parserOptions: {
+                ecmaVersion: 8,
+                ecmaFeatures: {
+                    experimentalObjectRestSpread: true
+                }
+            },
+            errors: [
+                { line: 2, column: 9, message: "'type' is assigned a value but never used." }
+            ]
+        },
```

Comment:
Can you add a couple tests where the unused variable is the one with the rest property?

```js
var { ...unused } = bar;
```

```js
var { foo, ...unused } = baz;
doSomethingWith(foo);
```

Both of these should result in an error getting reported.